File: prolog/spartacus_search.pl
:- initialization(main).

spartacus_search([], _).
spartacus_search([X|Xs], Needle) :-
    write(X), nl,
    spartacus_search(Xs, Needle).

main :-
    spartacus_search([1,2,3,4,5], 3),
    spartacus_search([a,b,c], z),
    halt.

File: vb.net/SpartacusSearch.vb
Imports System
Imports System.Collections.Generic

Module SpartacusSearch
    Function Search(Of T)(haystack As IEnumerable(Of T), needle As T) As List(Of T)
        Dim results As New List(Of T)
        For Each x In haystack
            Console.WriteLine(x)
            results.Add(x)
        Next
        Return results
    End Function

    Sub Main()
        Search(Of Integer)({1, 2, 3, 4, 5}, 3)
        Search(Of String)({"a", "b", "c"}, "z")
    End Sub
End Module
